src/lib/Nbe.ml

```diff
@@ -1018,15 +1018,24 @@ struct
       let* tm1 = branch_body phi1 clo1 in
       ret @@ S.CofSplit (ttp, tphi0, tphi1, tm0, tm1)
 
-  and quote_dim =
-    function
-    | D.Dim0 -> ret S.Dim0
-    | D.Dim1 -> ret S.Dim1
-    | D.DimVar lvl ->
+  (* stolen from quote_cof;we should replace those and other uses with calls to this, if this is the right thing *)
+  and quote_dim d = quote_con D.TpDim @@ D.dim_to_con d
+  (* this logis is right but ends up not being called because which variables are dimensions is lost;; todo/iev *)
+  (* function
+     | D.Dim0 -> ret S.Dim0
+     | D.Dim1 -> ret S.Dim1
+     | D.DimVar lvl ->
+     let* eq0 = lift_cmp @@ CmpM.test_sequent [] (Cof.eq (D.DimVar lvl) D.Dim0) in
+     let* eq1 = lift_cmp @@ CmpM.test_sequent [] (Cof.eq (D.DimVar lvl) D.Dim1) in
+     (* check to if the ctx can prove that the variable in question is either 0
+     or 1 before quoting just the level *)
+     if eq0 then ret S.Dim0
+     else if eq1 then ret S.Dim1
+     else
       let+ ix = quote_var lvl in
       S.Var ix
-    | D.DimProbe _ ->
-      failwith "DimProbe should not be quoted!"
+     | D.DimProbe _ ->
+     failwith "DimProbe should not be quoted!" *)
```

Comment:
Before merging, let's delete all this commented stuff.

Reply:
that commented stuff is still there because, also before merging, i'd like to take an hour or whatever to see if we can figure out the thing you mentioned with `Cut` instances that would allow us to print more helpful versions of dimension variables in hole output. that was also listed on #38 and i think falls under the general spirit of this PR.

Reply:
Oh, I forgot about that cut thing. Let me see if I can page it back in.

Reply:
<img width="415" alt="image" src="https://user-images.githubusercontent.com/55057/81095870-99ee5b00-8ed3-11ea-8b8b-cf11bfcba76f.png">


OK, this is my advice.

Reply:
good advice! worked a treat.